addons/binding/org.openhab.binding.keba/src/main/java/org/openhab/binding/keba/handler/KeContactBroadcastListener.java

```diff
@@ -0,0 +1,196 @@
+                ByteBuffer readBuffer = null;
+                InetSocketAddress clientAddress = null;
+
+                SelectionKey theSelectionKey = broadcastChannel.keyFor(broadcastListener);
+
+                if (theSelectionKey != null) {
+
+                    synchronized (broadcastListener) {
+                        try {
+                            broadcastListener.selectNow();
+                        } catch (IOException e) {
+                            logger.error("An exception occurred while selecting: {}", e.getMessage());
+                        }
+                    }
+
+                    Iterator<SelectionKey> it = broadcastListener.selectedKeys().iterator();
+                    while (it.hasNext()) {
+                        SelectionKey selKey = it.next();
+                        it.remove();
+                        if (selKey.isValid() && selKey.isReadable() && selKey == theSelectionKey) {
+
+                            readBuffer = ByteBuffer.allocate(KeContactHandler.BUFFER_SIZE);
+                            int numberBytesRead = 0;
+
+                            if (selKey == broadcastKey) {
+                                try {
+                                    clientAddress = (InetSocketAddress) broadcastChannel.receive(readBuffer);
+                                    logger.debug("Received {} from {} on the broadcast listener port ",
+                                            new String(readBuffer.array()), clientAddress);
+                                    numberBytesRead = readBuffer.position();
+
+                                } catch (Exception e) {
+                                    logger.error(
+                                            "An exception occurred while receiving data on the broadcast listener port: '{}'",
+                                            e.getMessage());
+                                }
+
+                            }
+
+                            if (numberBytesRead != -1) {
+                                readBuffer.flip();
+                            }
+                        }
+                    }
+                }
+
+                if (readBuffer != null && readBuffer.remaining() > 0) {
+                    Iterator<KeContactHandler> iterator = listeners.iterator();
+                    while (iterator.hasNext()) {
+                        KeContactHandler handler = iterator.next();
+                        if (clientAddress != null
+                                && handler.getIPAddress().equals(clientAddress.getAddress().getHostAddress())) {
+                            handler.onRead(readBuffer);
+                        }
+                    }
+                }
+            } catch (Exception e) {
+                logger.error(
+                        "An exception occurred while receiving broadcast data from a/the charging station(s) : '{}'",
+                        e.getLocalizedMessage(), e);
+            }
+        }
+    };
+}
```

Comment:
It is. The red-thing at the end of the line is gone now, only the comment is not properly marked outdated.